enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -0,0 +1,77 @@
+import type { Query } from "history";
+import { useState } from "react";
+
+import type { SdkClickActionPluginsConfig } from "embedding-sdk";
+import { InteractiveAdHocQuestion } from "embedding-sdk/components/private/InteractiveAdHocQuestion";
+import { getNewCardUrl } from "metabase/dashboard/actions/getNewCardUrl";
+import type { NavigateToNewCardFromDashboardOpts } from "metabase/dashboard/components/DashCard/types";
+import { useStore } from "metabase/lib/redux";
+import { getMetadata } from "metabase/selectors/metadata";
+import type { DashboardId, QuestionDashboardCard } from "metabase-types/api";
+
+import { StaticDashboard } from "../StaticDashboard";
+
+export type InteractiveDashboardProps = {
+  dashboardId: DashboardId;
+  initialParameterValues?: Query;
+  withTitle?: boolean;
+  withDownloads?: boolean;
+  hiddenParameters?: string[];
+  questionHeight?: number;
+  questionPlugins?: SdkClickActionPluginsConfig;
+};
+
+export const InteractiveDashboard = (props: InteractiveDashboardProps) => {
+  const { dashboardId, withTitle, questionHeight, questionPlugins } = props;
+
+  const store = useStore();
+
+  const [adhocQuestionUrl, setAdhocQuestionUrl] = useState<string | null>(null);
+
+  const handleNavigateToNewCardFromDashboard = ({
+    nextCard,
+    previousCard,
+    dashcard,
+    objectId,
+  }: NavigateToNewCardFromDashboardOpts) => {
+    const state = store.getState();
+    const metadata = getMetadata(state);
+    const { dashboards, parameterValues } = state.dashboard;
+    const dashboard = dashboards[dashboardId];
+
+    if (dashboard) {
+      const url = getNewCardUrl({
+        metadata,
+        dashboard,
+        parameterValues,
+        nextCard,
+        previousCard,
+        dashcard: dashcard as QuestionDashboardCard,
+        objectId,
+      });
+
+      if (url) {
+        setAdhocQuestionUrl(url);
```

Comment:
Maybe the problem is it's not even supposed to return `null` in the first place then? Given we always `dispatch(openUrl(url))` then it could be `dispatch(openUrl(null))`. There could be 2 scenarios here. It could either be
1. `getNewCardUrl` never return `null` in the first place
2. `getNewCardUrl` _can_ return `null` but `dispatch(openUrl(null))` doesn't break the app.

I think the intent isn't clear here, and I think we shouldn't make them look difference if the circumstances are the same. i.e. if embedding context do something differently and `getNewCardUrl` could be `null` then I'm happy with the solution you have here, but now it's a bit unclear what this part does to me.